### DISCUSSION TITLE
Mouse lagging.

### DISCUSSION THREAD

**Discussion by lorenzolamasse** [apohl79/audiogridder#913] [Help]
Hey !
Testing on latest beta release, win 10 server, mojave client, and the mouse seems to lag behind the stream, so i'ts pretty hard to use it without having values jumping all over the place, is it a known issue, or something specific to melda stuff ?
Keep up the good work !
![melda](https://user-images.githubusercontent.com/20107472/154636244-4319721c-036f-4087-b0c8-b8b4a6da1389.gif)

**Comment by oldtos** [comment]
I have the AG server on an old dual core with 6gb ram ( yea I know) But this isnt an issue...so Im poking in the dark..
On the sever when you use mouse on that GUI does it do the same thing? or just when you use the the plugin GUI?
Are you using a direct pc to pc lan connection or a gigabyte switch .. or a gigabit router ... with Ethernet cards capable of gigabyte speeds.. 
 (dont use wifi)
Is latency good sound wise....but only GUI lags?

**Comment by CaBuffalo** [comment]
What's your network set up?

Also, you might want to check your video card driver.

On Fri, Feb 18, 2022 at 7:45 PM oldtos ***@***.***> wrote:

> I have the AG server on an old dual core with 6gb ram ( yea I know) But
> this isnt an issue...so Im poking in the dark..
> On the sever when you use mouse on that GUI does it do the same thing? or
> just when you use the the plugin GUI?
> Are you using a direct pc to pc lan connection or a gigabyte switch .. or
> a gigabit router ... with Ethernet cards capable of gigabyte speeds..
> (dont use wifi)
> Is latency good sound wise....but only GUI lags?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/apohl79/audiogridder/discussions/913#discussioncomment-2208638>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ARMAR6HC3UNNFBSFQCYNSDTU34G57ANCNFSM5OXALZ6A>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***
> com>
>


-- 
Kevin Ma
Supported by Gmail.

**Comment by lorenzolamasse** [comment]
The server is in a VM inside the osx host so it's not a netwotk issue, packet delay is at 0. Graphic card is the one provided with the VirtualBox integration tools...

**Comment by CrossGridder** [comment]
What is the virtual display driver used by VirtualBox? If it is not SVGA3D, you might want to consider to try that.
I don't remember if it comes equipped in it, since SVGA3D is from VMWare. I think you can find it around and add it to VirtualBox if you really want to stick with that one, but there are better free VM apps in my opinion, such VMWare Fusion and UTM.

**Reply by lorenzolamasse** [reply]
These are the choice I get under osx: 
<img width="121" alt="image" src="https://user-images.githubusercontent.com/20107472/154801339-cf4055ca-2b1f-4cc5-9e6e-2bb072c41059.png">
SVGA3D doesn't seem to be proposed here. I'm trying to activate 3D graphics but shouldn't change anything...
Got some UIs not showing properly even inside the VM (VST2 and VST3) when 3D is acitvated, to I had to turn it off:
<img width="816" alt="image" src="https://user-images.githubusercontent.com/20107472/154801452-9a075c38-c0fd-49b0-a40c-26f806487302.png">
I'll try with UTM and report here.

**Comment by lorenzolamasse** [comment]
OK, mouse is lagging in the VM as well so that's most likely an issue of my GPU. Unfortunately UTM doesn't seem to be available for Mojave (or at least I haven't found which version is the latest compatible).
Meaning I'll probably have to update to Big Sur and maybe replace the GPU to something a bit more powerful, but it's clearly not the best time to purchase a GPU ;-)

**Reply by CrossGridder** [reply]
I've tried the same config as you on Monterey with an RX 580 8GB and the problem for me wasn't the GPU. The Virtual display driver is the issue. I had the lag and black GUI too until I started using SVGA3D. Until you don't use it, you will get the same issue also with NASA's GPUs.

**Comment by lorenzolamasse** [comment]
Switched to VMWare to compare, same issue, though the SVGA3D is installed, mouse is not lagging in the VM this time, but controlled plugins are when I try to tweak them, whether from the remote host or within the VM itself, so I'm getting back to some latency between what Audiogridder displays and the user actions...
